### DISCUSSION TITLE
Error in Github version

### DISCUSSION THREAD

**Discussion by aabouibrahim** [CheckMATE2/checkmate2#12] [General]
Dear Checkmate developers,

I downloaded the code from Github but it seems that the file Makefile.in is missing. Configure is failing with that error. A file called Makefile.am is present but this does not do the job.
Am I looking for the code in the wrong place?

Thank you,
Amin

**Comment by agagsgroove** [comment]
Hi Amin,

you have to generate the Makefile.in file.

Could you enter 

autoreconf

in the CM main folder and try to run configure again, please?

Cheers,

Jong

**Reply by aabouibrahim** [reply]
Hi Jong,

I mistakenly have been typing autoconf instead of autoreconf. 
All fixed now. CheckMate has been installed.

Thank you very much.

Best, Amin

**Comment by jjonesperez** [comment]
Hi, I have the same problem, but autoreconf ends in error:
```
tools/analysis/Makefile.am:10: error: Libtool library used but 'LIBTOOL' is undefined
tools/analysis/Makefile.am:10:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
tools/analysis/Makefile.am:10:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
tools/analysis/Makefile.am:10:   If 'LT_INIT' is in 'configure.ac', make sure
tools/analysis/Makefile.am:10:   its definition is in aclocal's search path.
autoreconf: automake failed with exit status: 1
```

I downgraded automake to v1.16.1, and managed generate the configure file using
```
autoconf
automake
```

After this, CheckMATE compiles without errors. However, when running AnalysisManager, I get the same message as above. I also get another error, which seems unrelated:
```
 Traceback (most recent call last):
  File "./AnalysisManager", line 8, in <module>
    reload(sys)
NameError: name 'reload' is not defined 
```
Any help would be appreciated!

**Comment by NishitaDesai** [comment]
Hi there, you also have to install the libtool package.

**Reply by jjonesperez** [reply]
Ah! Thanks, I hadn't noticed it was missing. Now things do compile, although I still get the same error with reload. Should I worry?

This seems to be a python2 vs python3 problem. I checked this solution:
https://stackoverflow.com/questions/961162/reloading-module-giving-nameerror-name-reload-is-not-defined

I tried modifying AnalysisManager, adding either
```
from six.moves import reload_module
reload_module(sys)
```
or
```
import importlib
importlib.reload(sys)
```

However, in both cases I get the output:
```
Traceback (most recent call last):
  File "/Collider/checkmate/bin/./AnalysisManager", line 11, in <module>
    import analysismanager_core
  File "/Collider/checkmate/tools/python/analysismanager_core.py", line 16
    print "    "*Printlevel + text
          ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("    "*Printlevel + text)?
```
Unsure if I should be worried about any of these messages, and if I can trust the CheckMATE output.

**Reply by jjonesperez** [reply]
Mmm, my student made me see that the analyses he designed were not incorporated with neither the "old" nor "new" versions of AnalysisManager.

What do you suggest? Downgrade to python2, and install a python2-friendly version of MadGraph?
